src/ServerResolver_nosrv.cpp

```diff
@@ -58,6 +64,8 @@ void ServerResolverPrivate::hostResolved(QHostInfo hostInfo) {
 		}
 
 		m_resolved << ServerResolverRecord(m_origHostname, m_origPort, 0, addresses);
+	} else {
+		m_lastError = ServerResolverError(hostInfo.error(), hostInfo.errorString());
 	}
 
 	emit resolved();
```

Comment:
We emit resovled when we failed to resolve? That seems wrong to me?

Reply:
The code in `TestServerResolver` waits until the signal is emitted (`QSignalSpy`), it checks for errors after that.